tests/create_skipped_test.go

```diff
@@ -0,0 +1,111 @@
+package tests
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/opera"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAccountCreation_CreateCallsWithInitCodesTooLargeDoNotAlterBalance(t *testing.T) {
+	versions := map[string]opera.Upgrades{
+		"sonic":   opera.GetSonicUpgrades(),
+		"allegro": opera.GetAllegroUpgrades(),
+	}
+
+	for name, version := range versions {
+		t.Run(name, func(t *testing.T) {
+			net := StartIntegrationTestNetWithJsonGenesis(t, IntegrationTestNetOptions{
+				Upgrades: &version,
+				ClientExtraArguments: []string{
+					"--disable-txPool-validation",
+				},
+			})
+
+			client, err := net.GetClient()
+			require.NoError(t, err)
+			defer client.Close()
+
+			sender := makeAccountWithBalance(t, net, big.NewInt(1e18))
+
+			gasPrice, err := client.SuggestGasPrice(t.Context())
+			require.NoError(t, err)
+
+			chainId, err := client.ChainID(t.Context())
+			require.NoError(t, err)
+
+			initCode := make([]byte, 50000)
+			txData := &types.LegacyTx{
+				Nonce:    0,
+				Gas:      10000000,
+				GasPrice: gasPrice,
+				To:       nil, // address 0x00 for contract creation
+				Value:    big.NewInt(0),
+				Data:     initCode,
+			}
+			tx := signTransaction(t, chainId, txData, sender)
+
+			preBalance, err := client.BalanceAt(t.Context(), sender.Address(), nil)
+			require.NoError(t, err)
+
+			receipt, err := net.Run(tx)
+			require.Error(t, err, "init code size too large")
```

Comment:
I think you might wanted to sue "require.ErrorContains" here.

The run above takes a long time until it times out. Maybe you can find a way to reduce this, as it adds 100s of seconds to the test time.